xsnippet_api/middlewares/auth.py

```diff
@@ -0,0 +1,47 @@
+"""
+    xsnippet_api.middlewares.auth
+    -------------------------------
+
+    The module implements auth middleware that performs authentication
+    of user requests.
+
+    :copyright: (c) 2016 The XSnippet Team, see AUTHORS for details
+    :license: MIT, see LICENSE for details
+"""
+
+from aiohttp import web
+from jose import jwt
+
+
+async def auth(app, next_handler):
+    """Authentication middleware
+
+    Performs user authentication by validating tokens passed in X-Auth-Token
+    header. Tokens are expected to be in JSON Web Token (RFC 7519) format.
+
+    On success the request object is modified to store an additional key 'auth'
+    which points to the parsed token value.
+
+    On failure 401 Unauthorized error is raised.
+
+    """
+
+    try:
+        secret = app['conf']['auth']['secret']
+    except KeyError:
+        raise ValueError("Option 'secret' is required to be set when "
+                         "auth middleware is enabled")
+
+    async def auth_handler(request):
+        token = request.headers.get('X-Auth-Token')
+        if token is not None:
+            try:
+                request['auth'] = jwt.decode(token, secret)
```

Comment:
Am I correct here that 

* `jwt.decode(...)` raises exception in case passed token is invalid.
* `jwt.decode(...)` returns human readable token including, for instance, user's permissions if it's valid, so we can check those in resources?